mypyc/primitives/set_ops.py

```diff
@@ -70,13 +72,12 @@ def emit_len(emitter: EmitterInterface, args: List[str], dest: str) -> None:
     error_kind=ERR_FALSE)
 
 # set.discard(obj)
-method_op(
+c_method_op(
     name='discard',
     arg_types=[set_rprimitive, object_rprimitive],
-    result_type=bool_rprimitive,
-    error_kind=ERR_FALSE,
-    emit=call_negative_bool_emit('PySet_Discard')
-)
+    return_type=c_int_rprimitive,
```

Comment:
The problem above(the magic error -11) is caused by this line of change, if we alias the `c_int_rprimtive` to the type represents 64 bit int(both `int64_t` or `Py_ssize_t` on a 64bit machine) would work. But changing it to `int32_t`  or int would fail.